### ISSUE TITLE
Implement dynamic number of worker threads

### ISSUE THREAD

**Issue by plajjan** [actonlang/acton#340]
The number of RTS worker threads is a very important factor to how an Acton system will act and perform.

Our current strategy is to spin up one worker thread per CPU core and pin these through CPU affinity.

We make an exception for smaller system with fewer than 4 CPU cores and then always create at least 4 worker threads.

I think it would be better for most systems and situations if we had a dynamic number of worker threads.

I don't want to remove the static strategy we have today. Rather I think we can add this new dynamic strategy (and make it default) but leave the static one in, so it is possible to select it with a command line argument. This makes it possible to experiment in the field. Though maybe the static config is really just a subset of dynamic config? If min / max workers have the same value, it will effectively be static.

- `--rts-workers`: sets a fixed number of workers
- `--rts-workers-min`: minimum number of workers; implies dynamic worker scaling
- `--rts-workers-max`: maximum number of workers; default to number of detected CPU cores
- `--rts-workers-load-max`: maximum number of workers for a system under load; that is, when 

- RTS starts out with `min_workers` number of workers
- workers pull jobs from the readyQ
- when a worker dequeues a job from the readyQ, it also peeks into the readyQ to see if there is anything there
  - if there is, it runs `ensure_wt()` whose purpose is to make sure there is another worker that can pick up that job
  - `ensure_wt()` will check if other worker threads are idle, then assume one of them will pick up the job
  -

**Comment by plajjan** [comment]
I should have written down the underlying reason to why I think dynamic number of threads, or a configurable amount of threads, would be useful; when I have experimented and hard-coded worker threads to certain numbers I have gotten significantly better performance. I have had a hunch that the performance characteristics of the run time system has been affected by races between threads and similar and thus the need for this issue. After #360 I am of a different idea. We have improved the worker loop in a way that can almost be described as fixing a bug. Performance is 10x and we have much lower syscall overhead etc.

The sort of bad behaviour we previously saw, why we might have wanted to adopt different scheduling strategies, is just gone. There might be reasons to still do dynamic worker threads; I'm thinking in particular when in overload scenarios and we want to give some form of pre-emptive scheduling or like fairness. That was never addressed in this issue and is for another issue I think.

Given all this, I don't think there is a point implementing this anymore as expressed here. We can gain experience with the characteristics of the new algorithm and if necessary open other issues.
